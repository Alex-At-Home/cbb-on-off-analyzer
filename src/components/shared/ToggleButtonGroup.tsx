// React imports:
import React, { useState } from 'react';


// Bootstrap imports:
import 'bootstrap/dist/css/bootstrap.min.css';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Row';
import Button from 'react-bootstrap/Button';
import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
import Tooltip from 'react-bootstrap/Tooltip';

export type ToggleButtonItem = {
  label: string | React.ReactNode,
  tooltip: string,
  toggled: boolean,
  onClick: () => void
}

type Props = {
  items: ToggleButtonItem[],
  override?: boolean //(for testing, set to true to force it to render)
}

const ToggleButtonGroup: React.FunctionComponent<Props> = ({items, override}) => {

  const tooltip = (tooltip: string, i: number) => <Tooltip id={`tooltip-${i}`}>{tooltip}</Tooltip>;

  return (override || (typeof window !== `undefined`)) ?
    <div><small>Quick Select:</small>&nbsp;&nbsp;
    {
      items.map((item, index) => {
          return <span key={"divtog" + index}>
            <OverlayTrigger placement="auto" overlay={tooltip(item.tooltip, index)}>
              <Button onClick={item.onClick} size="sm" key={"tog" + index} variant={item.toggled ? "dark" : "outline-secondary"}>{item.label}</Button>
            </OverlayTrigger>
            &nbsp;&nbsp;
          </span>
      })
    }</div> : <div/>; //(this construct needed to address SSR caching issue)
};
export default ToggleButtonGroup;
