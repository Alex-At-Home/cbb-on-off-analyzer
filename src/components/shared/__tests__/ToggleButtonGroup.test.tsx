import React from 'react';
import ToggleButtonGroup from '../ToggleButtonGroup';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';


describe("ToggleButtonGroup", () => {
  test("ToggleButtonGroup", () => {
    const component = renderer.create(
      <ToggleButtonGroup override={true} items={[
        {
          label: "Test A",
          tooltip: "Test Tooltip A",
          toggled: true,
          onClick: () => {}
        },
        {
          label: "Test B",
          tooltip: "Test Tooltip A",
          toggled: false,
          onClick: () => {}
        }
      ]}/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("ToggleButtonGroup - empty SSR render", () => {
    const component = renderer.create(
      <ToggleButtonGroup override={false} items={[
        {
          label: "Test A",
          tooltip: "Test Tooltip A",
          toggled: true,
          onClick: () => {}
        },
        {
          label: "Test B",
          tooltip: "Test Tooltip A",
          toggled: false,
          onClick: () => {}
        }
      ]}/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
