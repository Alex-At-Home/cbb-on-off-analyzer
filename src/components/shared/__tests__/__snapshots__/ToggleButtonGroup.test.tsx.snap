// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButtonGroup ToggleButtonGroup - empty SSR render 1`] = `<div />`;

exports[`ToggleButtonGroup ToggleButtonGroup 1`] = `
<div>
  <small>
    Quick Select:
  </small>
    
  <span>
    <div>
      <button
        className="btn btn-dark btn-sm"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Test A
      </button>
    </div>
      
  </span>
  <span>
    <div>
      <button
        className="btn btn-outline-secondary btn-sm"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Test B
      </button>
    </div>
      
  </span>
</div>
`;
